CLI support for Microsoft SecureScore report conversion (#2742)
mitre/saf: README.md

  USAGE
    $ saf convert msft_secure2hdf -r <secure-score-json> -p <secure-score-control-profiles> -o <hdf-scan-results-json> [-h]
    $ saf convert msft_secure2hdf -t <azure-tenant-id> -a <azure-app-id> -s <azure-app-secret> -o <hdf-scan-results-json> [-h]

  FLAGS
    -h, --help                                                  Show CLI help.
    -r, --inputScoreDoc=<secure-score-json>                     (required - mode 1) Input secureScore File
    -p, --inputProfiles=<secure-score-control-profiles-json>    (required - mode 1) Input secureScoreControlProfile File
    -t, --tenantId=<azure-tenant-id>                            (required - mode 2) Azure Tenant ID
    -a, --appId=<azure-app-id>                                  (required - mode 2) Azure App ID
    -s, --appSecreet=<azure-app-id>                             (required - mode 2) Azure App Secret
    -o, --output=<hdf-scan-results-json>                        (required) Output HDF JSON File

  EXAMPLES
    $ saf convert msft_secure2hdf -r secureScore.json -p secureScoreControlProfile.json -o output-hdf-name.json
    $ saf convert msft_secure2hdf -t "12345678-1234-1234-1234-1234567890abcd"   \

>                                   -p "12345678-1234-1234-1234-1234567890abcd"   \

Reviewer: ```suggestion
                                  -a "12345678-1234-1234-1234-1234567890abcd"   \
```

    $ saf convert nessus2hdf -i nessus_results.xml -o output-hdf-name.json
```

[top](#convert-other-formats-to-hdf)
#### Microsoft Secure Score to HDF
Output|Use|Command
---|---|---

> Msft Combined json|Combine the outputs from secureScore and secureScoreControlProfile | `convert msft_secure2hdf -i <(jq -s \'{"secureScore": .[0], "profiles": .[1]}\' secureScore.json secureScoreControlProfiles.json) -o output-hdf-name.json`

Reviewer: make this just the jq stuff, not the rest of the saf cli command

> Msft Combined json|Combine the outputs from secureScore and secureScoreControlProfile | `convert msft_secure2hdf -i <(jq -s \'{"secureScore": .[0], "profiles": .[1]}\' secureScore.json secureScoreControlProfiles.json) -o output-hdf-name.json`

Reviewer: add the commands necessary to use the api to pull down the securescore and profiles via curl or something.

output|use|command
---|---|---
microsoft secure score json|this file contains blah blah security results info|`curl something or other`
microsoft secure score control profiles json|this file contains all this metadata|`curl something else`
combined json|combines the outputs together into one file for ease of processing|`jq stuff`

> Msft Combined json|Combine the outputs from secureScore and secureScoreControlProfile | `convert msft_secure2hdf -i <(jq -s \'{"secureScore": .[0], "profiles": .[1]}\' secureScore.json secureScoreControlProfiles.json) -o output-hdf-name.json`

Reviewer: spell out microsoft

```

> convert msft_secure2hdf       Translate a Microsoft Secure Score report and Secure Score Control to a Heimdall Data Format JSON file. Required inputs:

Reviewer: get rid of the required inputs bit and the bullet points

                              * Pre-downloaded Microsoft Secure Score and Secure Score Control
                              Profiles JSON files
                              * Microsoft Graph API credentials

  USAGE
    $ saf convert msft_secure2hdf -r <secureScore-json> -p <secure-score-control-profiles> -o <hdf-scan-results-json> [-h]
    $ saf convert msft_secure2hdf -t <azure-tenant-id> -a <azure-app-id> -s <azure-app-secret> -o <hdf-scan-results-json> [-h]
    $ saf convert msft_secure2hdf -i <combined-inputs> -o <hdf-scan-results-json> [-h]

  FLAGS
    -h, --help                                                  Show CLI help.

>     -i, --combinedInputs                                        JSON incorporting {"secureScore": <-r>, "profiles": <-p>} File

Reviewer: ```suggestion
    -i, --combinedInputs                                        JSON File combining the outputs from the Microsoft Graph API endpoints
```

>     -r, --inputScoreDoc=<secure-score-json>                     Input secureScore File

Reviewer: ```suggestion
    -r, --inputScoreDoc=<secure-score-json>                     Input Secure Scores JSON File
```

>     -p, --inputProfiles=<secure-score-control-profiles-json>    Input secureScoreControlProfile File

Reviewer: ```suggestion
    -p, --inputProfiles=<secure-score-control-profiles-json>    Input Secure Score Control Profiles JSON File
```
